src/stdlib/Json.ts

```diff
@@ -0,0 +1,132 @@
+            return isInt32(x) ? new BrsInteger(x) : new BrsLongInteger(x);
+        }
+        return new BrsFloat(x);
+    case "object":
+        if (Array.isArray(x)) {
+            return new BrsArray(x.map(brsValueOf));
+        }
+        return new BrsAssociativeArray(
+            Object.getOwnPropertyNames(x).map((k: string) => {
+                return {
+                    name: brsValueOf(k),
+                    value: brsValueOf(x[k])
+                };
+            })
+        );
+    default:
+        throw new Error(`brsValueOf not implemented for: ${x} <${t}>`);
+    }
+}
+
+type ItemFn = (k: BrsString, v: BrsValue) => string;
+function itemsMap(brsAa: BrsAssociativeArray, fn: ItemFn) {
+    return brsAa.getElements().map((key: BrsString) => {
+        return fn(key, brsAa.get(key));
+    });
+}
+
+type ElementFn = (v: BrsJsonValue) => string;
+function elementsMap(brsArray: BrsArray, fn: ElementFn) {
+    return brsArray.getElements().map(fn);
+}
+
+function jsonOfItem(k: BrsString, v: BrsValue): string {
+    return `"${k.toString()}":${jsonOf(v)}`;
+}
+
+function jsonOf(x: BrsValue): string {
+    if (x instanceof BrsInvalid) {
+        return "null";
+    }
+    if (x instanceof BrsAssociativeArray) {
+        return `{${itemsMap(x, jsonOfItem).join(",")}}`;
+    }
+    if (x instanceof BrsArray) {
+        return `[${elementsMap(x, jsonOf).join(",")}]`;
+    }
+    if (x instanceof BrsString) {
+        return `"${x.toString()}"`;
+    }
+    if (!(x instanceof BrsUninitialized)) {
+        return x.toString();
+    }
+    throw new Error(`jsonValueOf not implemented for: ${x}`);
+}
+
+function logBrsErr(functionName: string, err: Error): void {
+    console.error(`BRIGHTSCRIPT: ERROR: ${functionName}: ${err.message}`);
+}
+
+export const FormatJson = new Callable("FormatJson", {
+    signature: { returns: ValueKind.String, args: [
+        { name: "x", type: ValueKind.Object },
+        {
+            name: "flags",
+            type: ValueKind.Int32,
+            defaultValue: new Literal(new BrsInteger(0))
+        }
+    ]},
+    impl: (_: Interpreter, x: BrsType, _flags: BrsInteger) => {
+        try {
+            return new BrsString(jsonOf(x));
+        } catch (err) {
+            // example RBI error:
+            // "BRIGHTSCRIPT: ERROR: FormatJSON: Value type not supported: roFunction: pkg:/source/main.brs(14)"
+            logBrsErr("FormatJSON", err);
+            return new BrsString("");
+        }
+    }
+});
+
+export const ParseJson = new Callable("ParseJson", {
+    signature: { returns: ValueKind.Dynamic, args: [
+        { name: "jsonString", type: ValueKind.String }
+    ]},
+    impl: (_: Interpreter, jsonString: BrsString) => {
+        try {
+            return brsValueOf(JSON.parse(jsonString.toString()));
+        } catch (err) {
+            // example RBI error:
+            // "BRIGHTSCRIPT: ERROR: ParseJSON: Unknown identifier 'x': pkg:/source/main.brs(25)"
```

Comment:
My approach was to use the same prefix (namely `"BRIGHTSCRIPT: ERROR: ParseJSON: "`) and implementation-defined error details. I think we'd have to write our own JSON parser rather than using the builtin TS/JS one. I suppose would could add various special cases to match specific known errors. I vote we add a ticket for this work as a nice-to-have.

Reply:
I'm not sure how `parseJson` and formatJson` would have access to the file and line number from the call site to add to the error message. Is that info available from the Interpreter?

Reply:
> I'm not sure how `parseJson` and `formatJson` would have access to the file and line number from the call site to add to the error message. Is that info available from the Interpreter?

At the moment, call-site information is stored in tokens only.  An instance of `Call` (from [Expression.ts](https://github.com/sjbarag/brs/blob/1e17f358debeb8f7ee3ff8d88f1551ac8bba2c30/src/parser/Expression.ts#L55)) contains the closing `)` token from the function call, but it's not passed here.  Not having call site locations isn't a deal breaker right now, though it does limit the usefulness of these error messages.  Feel free to file an issue for it!


> My approach was to use the same prefix (namely `"BRIGHTSCRIPT: ERROR: ParseJSON: "`) and implementation-defined error details. I think we'd have to write our own JSON parser rather than using the builtin TS/JS one. I suppose would could add various special cases to match specific known errors. I vote we add a ticket for this work as a nice-to-have.

Good point!  Would you mind special-casing the empty string message in this PR?  I don't expect to need many others.  A BrightScript dev may not have come across the JavaScript message for empty JSON strings before, and I was initially a bit concerned about malformed JSON before I realized what the JavaScript message meant.  Of course, empty strings _are_ malformed JSON, but "Data is empty" is pretty clear.  The other messages seem fine to use the JavaScript version for though!

Reply:
done

Reply:
Thanks!